frontend/app/actions/RegistrationUri.scala

```diff
@@ -33,10 +34,11 @@ object RegistrationUri {
     controllers.routes.Joiner.joinFriend().path -> "FRI"
   )
 
-  def parse(request: RequestHeader) = {
-    val redirectUrl: String = Config.idWebAppRegisterUrl(request.uri)
+  def parse(request: RequestHeader): String = {
     val campaignCode = extractCampaignCode(request.headers.get(REFERRER_HEADER), request.path)
-    redirectUrl.concat(s"&INTCMP=$campaignCode")
+    val returnUri: String = (request.uri ? ("INTCMP" -> campaignCode))
```

Comment:
I'd rename `returnUri` to `returnHereUrlAfterLogin` and `redirectUrl` to `identityRegistrationUrl` (or similar) otherwise it's hard to know what's going on as they both look too similar and not concrete